simpleviewinc/keg-components
# Modal style fixes

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-282)
> Should release package version 0.5.1 after this is merged

## Context
* Modal component wasn't working as intended
* Need to rework the Modal component with animation, styling
* also needed to update the storybook for modals for better looking examples

## Goal

* Add animations to default modal
* add better looking examples in storybook
* allow a bit more customization and control for the consumer

## Updates

* Added `@storybook/addon-knobs` for UI interaction testing
* updated the config to support knobs
* added a hook `useFromToAnimation` to make animating easier
* updated the styling for modals
* added an animationOut
* reworked the animation component so users can customize with their own components if they want
* added a few more optional props for hooking into animation callbacks

## Testing
* Run package
     * `keg dpg run package=docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:modal-style-fixes`
* check out the `Modal` section
* use the `knobs` to toggle modals in/out
